piecrust/src/session/call_stack.rs

```diff
@@ -0,0 +1,138 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use crate::instance::WrappedInstance;
+use std::collections::btree_map::Entry;
+
+use std::collections::BTreeMap;
+
+use piecrust_uplink::ModuleId;
+
+#[derive(Debug, Default)]
+pub struct CallStack {
+    // map of all instances together with a count in the stack.
+    instance_map: BTreeMap<ModuleId, (*mut WrappedInstance, u64)>,
```

Comment:
`CallStack` may drop without having released all the memory it contains. We should implement `Drop`